examples/pytorch/rgcn/entity_sample.py

```diff
@@ -1,167 +1,125 @@
+            logits = model(blocks)
+            loss = loss_fcn(logits, labels[output_nodes])
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        acc = evaluate(model, labels, val_loader, inv_target)
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} "
+              . format(epoch, total_loss / (it+1), acc))
+        
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='RGCN for entity classification with sampling')
-    parser.add_argument("--dropout", type=float, default=0,
-                        help="dropout probability")
-    parser.add_argument("--n-hidden", type=int, default=16,
-                        help="number of hidden units")
-    parser.add_argument("--gpu", type=int, default=0,
-                        help="gpu")
-    parser.add_argument("--n-bases", type=int, default=-1,
-                        help="number of filter weight matrices, default: -1 [use all]")
-    parser.add_argument("--n-epochs", type=int, default=50,
-                        help="number of training epochs")
-    parser.add_argument("-d", "--dataset", type=str, required=True,
-                        choices=['aifb', 'mutag', 'bgs', 'am'],
-                        help="dataset to use")
-    parser.add_argument("--wd", type=float, default=5e-4,
-                        help="weight decay")
-    parser.add_argument("--fanout", type=str, default="4, 4",
-                        help="Fan-out of neighbor sampling")
-    parser.add_argument("--use-self-loop", default=False, action='store_true',
-                        help="include self feature as a special relation")
-    parser.add_argument("--batch-size", type=int, default=100,
-                        help="Mini-batch size")
+    parser.add_argument("--dataset", type=str, default="aifb",
+                        help="Dataset name ('aifb', 'mutag', 'bgs', 'am').")
     args = parser.parse_args()
-
-    print(args)
-    main(args)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print(f'Training with DGL built-in RGCN module with sampling.')
+
+    # load and preprocess dataset
+    if args.dataset == 'aifb':
+        data = AIFBDataset()
+    elif args.dataset == 'mutag':
+        data = MUTAGDataset()
+    elif args.dataset == 'bgs':
+        data = BGSDataset()
+    elif args.dataset == 'am':
+        data = AMDataset()
+    else:
+        raise ValueError('Unknown dataset: {}'.format(args.dataset))
+    g = data[0]
+    num_rels = len(g.canonical_etypes)
+    category = data.predict_category
+    labels = g.nodes[category].data.pop('labels').to(device)
+    train_mask = g.nodes[category].data.pop('train_mask')
+    test_mask = g.nodes[category].data.pop('test_mask')
+    # find target category and node id
+    category_id = g.ntypes.index(category)
+    g = dgl.to_homogeneous(g, edata=None)
```

Comment:
Since the default value of `edata` is None, I think you can remove the specified value.

Reply:
Thanks. Fixed.